src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+
+        /*auto componentFlags = ref<uint64>(data_, 1);
+
+        if (componentFlags & )
+        {
+            // ...
+        }*/
+
+        char usernameBuffer[17] = {};
+        char passwordBuffer[17] = {};
+
+        std::memcpy(usernameBuffer, data_ + 0x09, 16);
+        std::memcpy(passwordBuffer, data_ + 0x19, 16);
+
+        // std::string reservedData(data_ + 0x40, 17);
+        std::string version(data_ + 0x51, 5);
+
+        std::string username(usernameBuffer, 16);
+        std::string password(passwordBuffer, 16);
+
+        if (strncmp(version.c_str(), SUPPORTED_XILOADER_VERSION, 5) != 0)
+        {
+            ref<uint8>(data_, 0) = LOGIN_ERROR_VERSION_UNSUPPORTED;
+
+            do_write(1);
+            return;
+        }
+
+        int8 code = ref<uint8>(data_, 0x29);
+
+        DebugSockets(fmt::format("auth code: {}", code));
+
+        // data check
+        if (check_string(username, 16) && check_string(password, 16))
+        {
+            ShowWarning("login_parse: send unreadable data");
+            return;
+        }
+        auto sql = std::make_unique<SqlConnection>();
+
+        char escaped_name[16 * 2 + 1] = {};
+        char escaped_pass[32 * 2 + 1] = {};
+
+        sql->EscapeString(escaped_name, username.c_str());
+        sql->EscapeString(escaped_pass, password.c_str());
+
+        username = escaped_name;
+        password = escaped_pass;
+
+        switch (code)
+        {
+            case LOGIN_ATTEMPT:
+            {
+                DebugSockets("LOGIN_ATTEMPT");
+
+                const char* fmtQuery = "SELECT accounts.id,accounts.status \
+                                    FROM accounts \
+                                    WHERE accounts.login = '%s' AND accounts.password = PASSWORD('%s')";
+                int32       ret      = sql->Query(fmtQuery, escaped_name, escaped_pass);
+                if (ret != SQL_ERROR && sql->NumRows() != 0)
+                {
+                    ret = sql->NextRow();
+
+                    uint32 accountID = sql->GetUIntData(0);
+                    uint32 status    = sql->GetUIntData(1);
+
+                    if (status & ACCOUNT_STATUS_CODE::NORMAL)
+                    {
+                        fmtQuery = "UPDATE accounts SET accounts.timelastmodify = NULL WHERE accounts.id = %d";
+                        sql->Query(fmtQuery, accountID);
+                        fmtQuery = "SELECT charid, server_addr, server_port \
+                                FROM accounts_sessions JOIN accounts \
+                                ON accounts_sessions.accid = accounts.id \
+                                WHERE accounts.id = %d;";
+                        ret      = sql->Query(fmtQuery, accountID);
+                        if (ret != SQL_ERROR && sql->NumRows() == 1)
+                        {
+                            while (sql->NextRow() == SQL_SUCCESS)
+                            {
+                                /*uint32 charid = sql->GetUIntData(0);
```

Comment:
dead code?

Reply:
Dead but intentionally left in if we ever want to restore `MSG_LOGIN` ZMQ. I can remove in the interim if wanted.